cli/src/cmd/forge/script/broadcast.rs

```diff
@@ -0,0 +1,260 @@
+                            deployment_sequence
+                                .save()
+                                .expect("not able to save deployment sequence");
+                            eyre::bail!("Aborting! A transaction failed to send: {:#?}", e)
+                        }
+                    };
+
+                    deployment_sequence.add_receipt(receipt);
+                    deployment_sequence.index += 1;
+                }
+                Err(e) => {
+                    deployment_sequence.save().expect("not able to save deployment sequence");
+                    eyre::bail!("{e}");
+                }
+            }
+        }
+
+        deployment_sequence.save()?;
+
+        println!("\n\n==========================");
+        println!(
+            "\nONCHAIN EXECUTION COMPLETE & SUCCESSFUL. Transaction receipts written to {:?}",
+            deployment_sequence.path
+        );
+        Ok(())
+    }
+
+    pub async fn handle_broadcastable_transactions(
+        &self,
+        target: &ArtifactId,
+        result: ScriptResult,
+        decoder: &mut CallTraceDecoder,
+        script_config: &ScriptConfig,
+    ) -> eyre::Result<()> {
+        if let Some(txs) = result.transactions {
+            println!("==========================");
+            println!("Simulated On-chain Traces:\n");
+            if let Ok(gas_filled_txs) = self.execute_transactions(txs, script_config, decoder).await
+            {
+                println!("\n\n==========================");
+                if !result.success {
+                    eyre::bail!("\nSIMULATION FAILED");
+                } else {
+                    let txs = gas_filled_txs;
+                    let mut deployment_sequence =
+                        ScriptSequence::new(txs, &self.sig, target, &script_config.config.out)?;
+                    deployment_sequence.save()?;
+
+                    if self.broadcast {
+                        self.send_transactions(&mut deployment_sequence).await?;
+                    } else {
+                        println!("\nSIMULATION COMPLETE. To broadcast these transactions, add --broadcast and wallet configuration(s) to the previous command. See forge script --help for more.");
+                    }
+                }
+            } else {
+                eyre::bail!("One or more transactions failed when simulating the on-chain version. Check the trace by re-running with `-vvv`")
+            }
+        } else if self.broadcast {
+            eyre::bail!("No onchain transactions generated in script");
+        }
+        Ok(())
+    }
+}
+
+pub fn set_chain_id(tx: &mut TypedTransaction, chain_id: u64) -> eyre::Result<()> {
+    match tx {
+        TypedTransaction::Legacy(tx) => tx.chain_id = Some(chain_id.into()),
+        TypedTransaction::Eip1559(tx) => tx.chain_id = Some(chain_id.into()),
+        _ => eyre::bail!("Wrong transaction type for expected output"),
+    }
+    Ok(())
+}
```

Comment:
given that this error will get caught in any of the other match branches, consider using this helper instead https://github.com/gakonst/ethers-rs/blob/master/ethers-core/src/types/transaction/eip2718.rs#L201-L208